contracts/src/v0.8/functions/dev/v1_X/FunctionsCoordinator.sol

```diff
@@ -142,27 +142,30 @@ contract FunctionsCoordinator is OCR2Base, IFunctionsCoordinator, FunctionsBilli
       report,
       (bytes32[], bytes[], bytes[], bytes[], bytes[])
     );
+    uint256 numberOfFulfillments = requestIds.length;
 
     if (
-      requestIds.length == 0 ||
-      requestIds.length != results.length ||
-      requestIds.length != errors.length ||
-      requestIds.length != onchainMetadata.length ||
-      requestIds.length != offchainMetadata.length
+      numberOfFulfillments == 0 ||
+      numberOfFulfillments != results.length ||
+      numberOfFulfillments != errors.length ||
+      numberOfFulfillments != onchainMetadata.length ||
+      numberOfFulfillments != offchainMetadata.length
     ) {
-      revert ReportInvalid();
+      revert ReportInvalid(
+        "All transmissions on the report must be equal: requestIds, results, errors, onchainMetadata, offchainMetadata"
+      );
     }
 
     // Bounded by "MaxRequestBatchSize" on the Job's ReportingPluginConfig
-    for (uint256 i = 0; i < requestIds.length; ++i) {
+    for (uint256 i = 0; i < numberOfFulfillments; ++i) {
```

Comment:
Let's not mix completely unrelated changes. Send out a separate PR for readability. Same for FunctionsBilling.sol.